unlock-app/src/components/interface/locks/Settings/forms/UpdateReferralFee.tsx

```diff
@@ -16,108 +22,157 @@ interface UpdateReferralFeeProps {
 
 interface FormProps {
   referralFeePercentage: number
+  referralAddress: string
 }
 
-const ZERO = ethers.constants.AddressZero
+const zeroAddress = ethers.constants.AddressZero
 
 export const UpdateReferralFee = ({
   lockAddress,
   network,
   isManager,
   disabled,
 }: UpdateReferralFeeProps) => {
-  const [referralFee, setReferralFee] = useState(false)
-  const web3Service = useWeb3Service()
-  const { getWalletService } = useAuth()
+  const [isReferralAddressToggled, setIsReferralAddressToggled] =
+    useState(false)
 
   const {
+    watch,
     register,
     handleSubmit,
+    reset,
     setValue,
     formState: { errors },
-  } = useForm<FormProps>()
-
-  const setReferrerFee = async (fields: FormProps) => {
-    const walletService = await getWalletService(network)
-    await walletService.setReferrerFee({
-      lockAddress,
-      address: ZERO,
-      feeBasisPoint: fields?.referralFeePercentage * 100,
-    })
-  }
+  } = useForm<FormProps>({
+    defaultValues: { referralFeePercentage: 0, referralAddress: '' },
+  })
 
-  const getReferrerFees = async () => {
-    return await web3Service.referrerFees({
-      lockAddress,
-      network,
-      address: ZERO,
-    })
-  }
-
-  const setReferrerFeeMutation = useMutation(setReferrerFee)
+  const {
+    data: referralFees,
+    isLoading,
+    isSettingReferrerFee,
+    setReferrerFee,
+  } = useReferrerFee({
+    lockAddress,
+    network,
+  })
 
   const onSubmit = async (fields: FormProps) => {
-    const setReferrerFeePromise = setReferrerFeeMutation.mutateAsync(fields)
+    const setReferrerFeePromise = setReferrerFee.mutateAsync({
+      ...fields,
+      referralAddress: referralAddress || zeroAddress,
+    })
```

Comment:
This will apply to the address you've entered or apply the address to "any" referrer. A.k.a `0x0000000000000000000000000000000000000000`